packages/react/src/firestore/useGetAggregateFromServerQuery.ts

```diff
@@ -0,0 +1,41 @@
+import { useQuery, type UseQueryOptions } from "@tanstack/react-query";
+import {
+  type Query,
+  type FirestoreError,
+  getAggregateFromServer,
+  type AggregateSpec,
+  type AggregateField,
+} from "firebase/firestore";
+
+type FirestoreUseQueryOptions<TData = unknown, TError = Error> = Omit<
+  UseQueryOptions<TData, TError>,
+  "queryFn"
+> & {
+  firestore: {
+    aggregateSpec: AggregateSpec;
+  };
+};
+
+type AggregateResult<T extends AggregateSpec> = {
+  [K in keyof T]: AggregateField<T[K]> extends AggregateField<infer R>
+    ? R
+    : never;
+};
+
+export function useGetAggregateFromServerQuery<T extends AggregateSpec>(
```

Comment:
I think we should change the API spec here, since the aggregates are a requirement, so:

```
useGetAggregateFromServerQuery(query, {
    countOfDocs: count(),
    totalHours: sum('hours'),
    averageScore: average('score')
});
```

Let's keep any optional parts in the firestore key and make sure they are all optional values?

Reply:
thanks for the clarity here. absolutely, going to adjust here and keep all required parts at the top level and optional parts maintaining them as optional in the firestore key object